doc/api/n-api.md

```diff
@@ -3155,6 +3156,81 @@ support it:
+that each action hinges on the outcome of the previous action.
+
+N-API implements promises as a pair of two objects - a "deferred" and a
+promise. A "deferred" is a JavaScript value created along with the promise by
+`napi_create_promise()`. The promise can be passed into JavaScript while the
+deferred must be retained on the native side, because the promise that has been
+derived from it can only be resolved or rejected by passing the deferred into
+`napi_conclude_deferred()`.
+
+### napi_create_promise
+<!-- YAML
+added: REPLACEME
+-->
+```C
+NAPI_EXTERN napi_status napi_create_promise(napi_env env,
+                                            napi_value* deferred,
+                                            napi_value* promise);
+```
+
+- `[in] env`: The environment that the API is invoked under.
+- `[out] deferred`: The deferred object.
+- `[out] promise`: The JavaScript promise associated with the deferred object.
+
+Returns `napi_ok` if the API succeeded.
+
+This API creates a deferred object and a JavaScript promise. The two objects
+are linked such that the deferred can later be passed to
+`napi_conclude_deferred()` to resolve the promise. Although the deferred object
+is a JavaScript value it should not be passed into JavaScript because it lacks
+the functionality of a JavaScript promise. However, the JavaScript promise
+returned in the third parameter can be passed into JavaScript.
+
+### napi_conclude_deferred
+<!-- YAML
+added: REPLACEME
+-->
+```C
+NAPI_EXTERN napi_status napi_conclude_deferred(napi_env env,
+                                               napi_value deferred,
+                                               napi_value conclusion,
+                                               bool is_resolution);
+```
+
+- `[in] env`: The environment that the API is invoked under.
+- `[in] deferred`: The deferred object whose associated promise to conclude.
+- `[in] conclusion`: The value with which to conclude the promise.
+- `[in] is_rejection`: Flag indicating whether `conclusion` is a resolution
+(`true`) or a rejection (`false`).
+
+This API resolves or rejects a JavaScript promise by way of the deferred object
+from which it was derived. Thus, it can only be used to conclude JavaScript
+promises for which the corresponding deferred object is available. This
+effectively means that the promise must have been created using
+`napi_create_promise()` and the deferred object returned from that call must
+have been retained in order to be passed to this API.
+
+### napi_is_promise
+<!-- YAML
+added: REPLACEME
+-->
+```C
+NAPI_EXTERN napi_status napi_is_promise(napi_env env,
+                                        napi_value promise,
+                                        bool* is_promise);
+```
+
+- `[in] env`: The environment that the API is invoked under.
+- `[in] promise`: The promise to examine
+- `[out] is_promise`: Flag indicating whether `promise` is a native promise
```

Comment:
@gabrielschulhof I mean I'm fine with `is_promise` only caring about and working with native promises - and ignoring non-native promises in n-api completely